### PR TITLE
Teams mention

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19813

## Description
Added the ability to mention users when messaging to Teams.

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (with link to it)
- [x] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/blob/85f0261ba57b616f4e6c79f7f14e8bd80127da74/Integrations/MicrosoftTeams/MicrosoftTeams.yml) lines 106-107 (description of send-notification command)
- [ ] [CHANGELOG](https://github.com/demisto/content/blob/85f0261ba57b616f4e6c79f7f14e8bd80127da74/Integrations/MicrosoftTeams/CHANGELOG.md)

### CODE REVIEW

**Comment by Itay4** [comment]
Not big fan of the ; char, do we really need it?

**Comment by guyfreund** [comment]
@Itay4 Usually users are "FirstName LastName" with a space in between, we need a mark to know the end of a mention

File: Integrations/MicrosoftTeams/MicrosoftTeams_test.py [hunk 1]
```diff
@@ -100,6 +100,41 @@ def get_bot_access_token(requests_mock):
...
+    assert mentioned_users_to_entities(mentioned_users, integration_context) == [bruce_entity, denzel_entity]
+
+    mentioned_users = ['Bruce Willis', 'demisto']
+    with pytest.raises(ValueError, match='Team member demisto was not found'):
+        mentioned_users_to_entities(mentioned_users, integration_context)
+
+
+def test_process_mentioned_users_in_message():
+    from MicrosoftTeams import process_mentioned_users_in_message
+    raw_message = '@demisto dev; @demisto; email@demistodev.com; demisto@demistodev.com'
```
Comment (ShahafBenYakir) [inline Integrations/MicrosoftTeams/MicrosoftTeams_test.py]: Add a case where a `@` was given with no trailing `;` and the opposite case, where trailing `;` was given with no leading `@`

File: Integrations/MicrosoftTeams/MicrosoftTeams.py [hunk 2]
```diff
@@ -969,6 +970,31 @@ def send_message_request(service_url: str, channel_id: str, conversation: dict):
...
+
+
+def mentioned_users_to_entities(mentioned_users: list, integration_context: dict) -> list:
+    """
+    Returns a list of entities built from the mentioned users
+    :param mentioned_users: A list of mentioned users in the message
+    :param integration_context: Cached object to retrieve relevant data from
+    :return: A list of entities
+    """
+    return [{'type': 'mention', 'mentioned': {'id': get_team_member_id(user, integration_context), 'name': user},
```
[the remark this answers is not in the document]
Comment (guyfreund) [reply Integrations/MicrosoftTeams/MicrosoftTeams.py]: Nope, id field is a mandatory in user object

File: Integrations/MicrosoftTeams/MicrosoftTeams.yml [hunk 3]
```diff
@@ -102,7 +102,9 @@ script:
       required: false
       secret: false
     deprecated: false
-    description: Sends a message to the specified teams.
+    description: |-
+      Sends a message to the specified teams.
+      If you want to mention a user please make sure to add ";" at the end of the mention. For example: @Bruce Willis;
```
Comment (kirbles19) [inline Integrations/MicrosoftTeams/MicrosoftTeams.yml]: To mention a user in the message, add a semicolon ";" at the end of the user mention. For example: @JohnDoh;

Comment (guyfreund) [reply Integrations/MicrosoftTeams/MicrosoftTeams.yml]: @kirbles19  We want them to know that a user can be assembled from one or more words, so it's better to give a full name for example. Is Bruce Willis a problem?

**Comment by kirbles19** [comment]
@guyfreund See comment inline. All done.

**Pull request merged**